### PR TITLE
Add PyPI publish support

### PR DESCRIPTION
**Pull request by KedoKudo**
# Short description of the changes:

Add files and instructions on PyPI publication.

# Long description of the changes:

This PR introduces the following changes:

- Add missing required entries in `pyproject.toml` for PyPI publishing.
- Add additional deps.
- Add new workflow target for publishing to PyPI.
- Add instructions on PyPI local build, test and publishing to testPyPI

# Check list for the pull request
- [x] I have read the [CONTRIBUTING]
- [x] I have read the [CODE_OF_CONDUCT]
- [x] I have added tests for my changes
- [x] I have updated the documentation accordingly

# Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

# Manual test for the reviewer

N/A

# References

Related EWM item: [Task5917](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=5917)

### CODE REVIEW

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -45,3 +45,28 @@ jobs:
           if [ "${IS_RC}" = "true" ]; then CONDA_LABEL="rc"; fi
           echo pushing ${{ github.ref }} with label $CONDA_LABEL
           anaconda upload --label $CONDA_LABEL conda.recipe/noarch/mypackagename*.tar.bz2
+
+  pypi-publish:
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v')
```
Comment (peterfpeterson) [inline .github/workflows/package.yml:52]: Can we move this "if" to just the publish step? That way you always find out if the package builds

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -45,3 +45,28 @@ jobs:
...
+          channels: conda-forge,defaults
+          mamba-version: "*"
+          environment-file: environment.yml
+          cache-environment-key: ${{ runner.os }}-env-${{ hashFiles('**/environment.yml') }}
+          cache-downloads-key: ${{ runner.os }}-downloads-${{ hashFiles('**/environment.yml') }}
+      - name: build pypi pacakge
+        run: |
+          # build the package
+          VERSION=$(versioningit .) python -m build
+      # retrieve your distributions here
```
Comment (peterfpeterson) [inline .github/workflows/package.yml:70]:
```suggestion
      # publish your distributions here
```

File: .github/workflows/package.yml (hunk 1 above)
Comment (KedoKudo) [reply .github/workflows/package.yml:52]: Move the if check to be final step only.

File: .github/workflows/package.yml (hunk 2 above)
Comment (KedoKudo) [reply .github/workflows/package.yml]: fixed with latest commit.

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**